### PR TITLE
Implemented simple Tax Calculator

### PR DESCRIPTION
**Pull request by pawkli95**

### CODE REVIEW

**Review by katdunaj — APPROVED** [review]

File: app/src/main/java/pl/futurecollars/invoicing/utils/FileService.java [hunk 1]
```diff
@@ -2,15 +2,24 @@
...
 
+@Slf4j
 public class FileService {
 
     private final File file;
 
     public FileService(String fileName) {
         file = new File(fileName);
+        try {
+            Files.createFile(Paths.get(fileName));
```
Comment (radoslawdabrowski) [inline app/src/main/java/pl/futurecollars/invoicing/utils/FileService.java:19]: Actually, if we have problem with creating a file this service should not be initialized, so it is better to throw some implementation of RuntimeException (or even checked Exception) than try catch in constructor.

**Review by radoslawdabrowski — CHANGES_REQUESTED** [review]

File: app/src/main/java/pl/futurecollars/invoicing/utils/FileService.java [hunk 2]
```diff
@@ -2,15 +2,27 @@
...
+        } catch (FileAlreadyExistsException e) {
+            log.info(e.getMessage());
+        } catch (IOException e) {
+            throw new RuntimeException();
```
Comment (radoslawdabrowski) [inline app/src/main/java/pl/futurecollars/invoicing/utils/FileService.java:24]: "implementation of RuntimeException" :) create e.g. FileNotCreatedException that extends RuntimeException and throw it here

**Review by radoslawdabrowski — CHANGES_REQUESTED** [review]

**Review by katdunaj — DISMISSED** [review]

**Review by michalflis — COMMENTED** [review]
In TaxCalculatorServiceIntegrationTest.groovy and TaxCalculatorServiceUnitTest.groovy you are testing the same, is it necessary?

**Comment by pawkli95** [comment]
> 
> 
> In TaxCalculatorServiceIntegrationTest.groovy and TaxCalculatorServiceUnitTest.groovy you are testing the same method, is it necessary?

I think so, in the unit test I tested only the method by mocking the database, and in integrationTest database has real implementation to test the whole module

**Review by michalflis — APPROVED** [review]

**Review by katdunaj — APPROVED** [review]

**Review by radoslawdabrowski — APPROVED** [review]

**Pull request merged**